presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcResultSetTimezone.java

```diff
@@ -0,0 +1,381 @@
+        assertParameter(sameInstantInWarsawZone, sessionTimezoneId, (ps, i) -> ps.setTimestamp(i, sqlTimestamp, Calendar.getInstance(TimeZone.getTimeZone("Europe/Warsaw"))));
+        assertParameter(sqlTimestamp, sessionTimezoneId, (ps, i) -> ps.setObject(i, sqlTimestamp));
+        assertParameter(new Timestamp(sqlDate.getTime()), sessionTimezoneId, (ps, i) -> ps.setObject(i, sqlDate, Types.TIMESTAMP));
+        assertParameter(new Timestamp(sqlTime.getTime()), sessionTimezoneId, (ps, i) -> ps.setObject(i, sqlTime, Types.TIMESTAMP));
+        assertParameter(sqlTimestamp, sessionTimezoneId, (ps, i) -> ps.setObject(i, sqlTimestamp, Types.TIMESTAMP));
+        assertParameter(sqlTimestamp, sessionTimezoneId, (ps, i) -> ps.setObject(i, javaDate, Types.TIMESTAMP));
+        assertParameter(sqlTimestamp, sessionTimezoneId, (ps, i) -> ps.setObject(i, dateTime, Types.TIMESTAMP));
+        assertParameter(sqlTimestamp, sessionTimezoneId, (ps, i) -> ps.setObject(i, "2001-05-06 12:34:56", Types.TIMESTAMP));
+    }
+
+    private void assertParameter(Object expectedValue, String sessionTimezoneId, Binder binder)
+            throws SQLException
+    {
+        connection.unwrap(PrestoConnection.class).setTimeZoneId(sessionTimezoneId);
+        try (PreparedStatement statement = connection.prepareStatement("SELECT ?")) {
+            binder.bind(statement, 1);
+
+            try (ResultSet rs = statement.executeQuery()) {
+                assertTrue(rs.next());
+                assertEquals(expectedValue, rs.getObject(1));
+                assertFalse(rs.next());
+            }
+        }
+    }
+
+    private void checkRepresentation(String expression, String sessionTimezoneId, ResultAssertion assertion)
+            throws Exception
+    {
+        for (ReferenceDriver driver : referenceDrivers) {
+            // log the name of the driver if we're going to use more than one
+            checkRepresentation(expression, sessionTimezoneId, driver, assertion);
+        }
+    }
+
+    private void checkRepresentation(String expression, String sessionTimezoneId, ReferenceDriver reference, ResultAssertion assertion)
+            throws Exception
+    {
+        try (ResultSet rs = prestoQuery(expression, sessionTimezoneId); ResultSet referenceResultSet = reference.query(expression, sessionTimezoneId)) {
+            assertTrue(rs.next());
+            assertTrue(referenceResultSet.next());
+            assertion.accept(rs, referenceResultSet, 1);
+            assertFalse(rs.next());
+            assertFalse(referenceResultSet.next());
+        }
+    }
+
+    private ResultSet prestoQuery(String expression, String timezoneId)
+            throws Exception
+    {
+        connection.unwrap(PrestoConnection.class).setTimeZoneId(timezoneId);
+        return statement.executeQuery("SELECT " + expression);
+    }
+
+    private Calendar getCalendar()
+    {
+        return Calendar.getInstance(TimeZone.getTimeZone("Asia/Kathmandu"));
```

Comment:
Is the timezone returned by this method meaningful (the assertions test should pass - no matter which one we return here). If for some reason we want it to match JVM timezone please add an assertion that it actually does. 

Or maybe we have such assertion somewhere globally? I vaguely remember that we may have such.

Reply:
Nothing special, it's just a timezone that's **different** from the JVM timezone (America/Bahia_Banderas).  Maybe an assert that the 2 are different would be useful.  I'll also add a comment.

Reply:
O yeah my bad. We are using specifically `Asia/Kathmandu` too. But i think it is used as hive session time zone.

Comment and assertion will be helpful. Thanks.